src/entity/find-entities.ts

```diff
@@ -88,43 +88,36 @@ export async function findEntities(
     throw visitError;
   }
 
-  return orderEntities(document, nestedEntities);
+  return orderEntities(nestedEntities);
 }
 
-function orderEntities(
-    document: ParsedDocument<any, any>,
-    unorderedEntities: ScannedFeature[][]): ScannedFeature[] {
-  // Build a map of node -> entities
-  let entitiesByNode = new Map<any, ScannedFeature[]>();
-  for (let entitySet of unorderedEntities) {
-    for (let entity of entitySet) {
-      let node = entity.node || null;  // convert undefined to null
-      let entities = entitiesByNode.get(node);
-      if (!entities) {
-        entities = [];
-        entitiesByNode.set(node, entities);
-      }
-      entities.push(entity);
-    }
+function compareEntitiesBySourceLocation(
+    ent1: ScannedFeature, ent2: ScannedFeature): number {
+  const range1 = ent1.sourceRange;
+  const range2 = ent2.sourceRange;
+  if (range2 == null) {
+    // ent1 comes first
+    return -1;
```

Comment:
I think you want to handle the both null case first so you return 0.

```
if (range1 == range2) return 0;
if (range1 == null) return -1;
if (range2 == null) return 1;
```

btw, this is a case where single-line if's are nice. Our style guide allows it :)